### PR TITLE
feat: add blog post about fixing code coverage gaps

### LINKED ISSUE #323: 🚀 Feature: Blog post on resolving test code coverage gaps
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/dot-com/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I end up leaving these comments a _lot_ on PRs. For each gap in coverage, at least one of the following is probably true:

* The code is reachable, so a unit test should exercise it
* The code can be refactored to not include that case
* This is a difficult-to-represent edge case in types and a type assertion is warranted

### Additional Info

Oftentimes when I post these it's on a lint rule that needs more `RuleTester` testing. But I don't want the version on my blog to be specific to lint rules - that stuff is intimidating to many. I'll probably suggest an equivalent blog post on the typescript-eslint blog for specifically lint rules.

💖

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #323
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/dot-com/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/dot-com/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Answers the three common situations, with examples. Also adds an encouraging note at the end.

Shoutout @dimitropoulos (@michigantypescript) for the ending image.

### CODE REVIEW

File: src/content/blog/so-youve-got-a-gap-in-code-coverage/index.mdx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+// 🛑 See later - there's a better way!
+const count = (counts.get(value) ?? 0) + 1;
+```
+
+```ts
+const count = counts.get(value)! + 1;
+```
+
+> 🚫 Type assertions are almost always a last ditch resort.
+> They're still better than using `any`, or even more unsafe, [TypeScript comment directives](https://learningtypescript.com/articles/comment-directives).
```
Comment (JoshuaKGoldberg) [inline src/content/blog/so-youve-got-a-gap-in-code-coverage/index.mdx]: Blocked on https://github.com/LearningTypeScript/site/issues/142.

Comment (JoshuaKGoldberg) [reply src/content/blog/so-youve-got-a-gap-in-code-coverage/index.mdx]: ✅
